CMakeLists.txt

```diff
@@ -3,8 +3,44 @@ cmake_minimum_required(VERSION 3.0)
 project(libuv)
 enable_testing()
 
+if(DEFINED BUILD_SHARED_LIBS)
+  if(BUILD_SHARED_LIBS)
+    if(DEFINED LIBUV_ENABLE_SHARED AND NOT LIBUV_ENABLE_SHARED)
+      message(WARNING "LIBUV_ENABLE_SHARED is overridden to ON by BUILD_SHARED_LIBS")
+    endif()
+    set(LIBUV_ENABLE_SHARED ON)
+    if(DEFINED LIBUV_ENABLE_STATIC AND LIBUV_ENABLE_STATIC)
+      message(WARNING "LIBUV_ENABLE_STATIC is overridden to OFF by BUILD_SHARED_LIBS")
+    endif()
+    set(LIBUV_ENABLE_STATIC OFF)
+  else()
+    if(DEFINED LIBUV_ENABLE_SHARED AND LIBUV_ENABLE_SHARED)
+      message(WARNING "LIBUV_ENABLE_SHARED is overridden to OFF by BUILD_SHARED_LIBS")
+    endif()
+    set(LIBUV_ENABLE_SHARED OFF)
+    if(DEFINED LIBUV_ENABLE_STATIC AND NOT LIBUV_ENABLE_STATIC)
+      message(WARNING "LIBUV_ENABLE_STATIC is overridden to ON by BUILD_SHARED_LIBS")
+    endif()
+    set(LIBUV_ENABLE_STATIC ON)
+  endif()
+else()
+  if(NOT DEFINED LIBUV_ENABLE_SHARED)
+    set(LIBUV_ENABLE_SHARED ON)
+  endif()
+  if(NOT DEFINED LIBUV_ENABLE_STATIC)
+    set(LIBUV_ENABLE_STATIC ON)
+  endif()
```

Comment:
Do libuv's other build systems support building static and shared libraries simultaneously?

Doing so here adds a lot of complexity.  I'd prefer to see a PR that does all the other fixes without considering this extension in functionality yet.

Reply:
@bradking I deleted the function of choosing whether to build a static library or a dynamic library. Since  #1886 was created separately, I removed Windows installation etc.

> Do libuv's other build systems support building static and shared libraries simultaneously?

We can select the library to build with `autogen.sh`. Also, with `gyp` it can be selected exclusively.